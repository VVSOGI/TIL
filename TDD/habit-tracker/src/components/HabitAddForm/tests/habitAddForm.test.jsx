import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import HabitAddForm from '../habitAddForm';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

describe('HabitAddForm', () => {
  let onAdd;
  let input;
  let button;
  it('renders', () => {
    const component = render(<HabitAddForm onAdd={jest.fn()} />);
    expect(component.container).toMatchSnapshot();
  });

  describe('Form Submit', () => {
    beforeEach(() => {
      onAdd = jest.fn();
      render(<HabitAddForm onAdd={onAdd} />);
      input = screen.getByPlaceholderText('Habit');
      button = screen.getByText('Add');
    });

    it('calls onAdd when button is clicked and valid habit is entered', () => {
      userEvent.type(input, 'New Habit');
      userEvent.click(button);

      expect(onAdd).toHaveBeenCalledWith('New Habit');
    });

    it('does not call onAdd when the habit is empty', () => {
      userEvent.type(input, '');
      userEvent.click(button);

      expect(onAdd).toHaveBeenCalledTimes(0);
    });
  });
});
