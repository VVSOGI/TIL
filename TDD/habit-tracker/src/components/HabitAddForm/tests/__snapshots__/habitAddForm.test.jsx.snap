// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitAddForm renders 1`] = `
<div>
  <form
    class="add-form"
  >
    <input
      class="add-input"
      placeholder="Habit"
      type="text"
    />
    <button
      class="add-button"
    >
      Add
    </button>
  </form>
</div>
`;
